code/modules/cargo/department_order.dm

```diff
@@ -114,6 +114,8 @@ GLOBAL_LIST_INIT(department_order_cooldowns, list(
 
 	. = TRUE
 	var/id = text2path(params["id"])
+	if(!ispath(id, /datum/supply_pack)) // this means its not a valid path, used only in the case of dynamically generated canister packs
+		id = params["id"]
```

Comment:
This feels like an icky solution, Maybe just remove the text2path and validate it's a supply pack from there (which is already done in the next line)?

Alternatively, adjust the way supply packs are fed into `ui_data` to accommodate for dynamic packs?

Reply:
So uh after some testing, I could not find a way to get `ui_static_data` to accommodate for dynamic packs without a similar solution, and ispath(params["id"]) just always returns false due to how byond works.

Reply:
If not all IDs are typepaths, then maybe get rid of the `text2path` all together. Evidently, some IDs are not paths.